lib/Workflow/Persister.pm

```diff
@@ -53,11 +53,14 @@ sub assign_generators {
 
 sub init_random_generators {
     my ( $self, $params ) = @_;
-    my $length = $params->{id_length} || DEFAULT_ID_LENGTH;
-    eval { require Workflow::Persister::RandomId };
-    if (my $msg = $EVAL_ERROR) {
+    my $length  = $params->{id_length} || DEFAULT_ID_LENGTH;
+    try {
+        require Workflow::Persister::RandomId;
+    }
+    catch ($msg) {
         $msg =~ s/\\n/ /g;
         $self->log->error($msg);
+        ###TODO: shouldn't the error be rethrown?? the code below happily tries to use the failed-to-load class!
```

Comment:
That sound reasonable, perhaps we could implement a test ?

Reply:
It's easy to rethrow, but it's hard to test: we'd need to make loading of `Workflow::Persister::RandomId` fail with an exception. I'll investigate what level of trickery I can come up with to test the "catch" branch of this.

Reply:
Test::Without::Module could be an option. Checking that.